packages/flutter_tools/bin/macos_assemble.sh

```diff
@@ -118,6 +118,7 @@ BuildApp() {
     "-dDarwinArchs=${ARCHS}"
     "-dTargetFile=${target_path}"
     "-dBuildMode=${build_mode}"
+    "-dFlavor=${FLAVOR}"
```

Comment:
is sending `-dFlavor= ` (empty string) equivalent to not even passing the dart define?

Reply:
since we only conditionally add this in xcode_backend.dart: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/bin/xcode_backend.dart#L404

Reply:
i see we also have conditional flags in this file starting at line 135

Reply:
although, I also see that MOST of the flags in xcode_backend.dart that read from the environment fall back to empty string.

Reply:
oh wait, actually, these aren't even dart defines, but Xcode compiler defines, right?